contracts/verifiers/Verifier.sol

```diff
@@ -0,0 +1,688 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.27;
+
+import {IERC165} from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import {ContextUpgradeable} from "@openzeppelin/contracts-upgradeable/utils/ContextUpgradeable.sol";
+import {IRequestValidator} from "../interfaces/IRequestValidator.sol";
+import {IAuthValidator} from "../interfaces/IAuthValidator.sol";
+import {IState} from "../interfaces/IState.sol";
+import {VerifierLib} from "../lib/VerifierLib.sol";
+import {IVerifier} from "../interfaces/IVerifier.sol";
+
+error RequestIdNotFound(uint256 requestId);
+error RequestAlreadyExists(uint256 requestId);
+error GroupIdNotFound(uint256 groupId);
+error GroupIdAlreadyExists(uint256 groupId);
+error MultiRequestIdNotFound(uint256 multiRequestId);
+error MultiRequestIdAlreadyExists(uint256 multiRequestId);
+error AuthTypeNotFound(string authType);
+error AuthTypeAlreadyExists(string authType);
+error ValidatorNotWhitelisted(address validator);
+error RequestIsAlreadyGrouped(uint256 requestId);
+error LinkIDNotTheSameForGroupedRequests(uint256 requestLinkID, uint256 requestLinkIDToCompare);
+error UserIDNotFound(uint256 userID);
+error UserIDNotLinkedToAddress(uint256 userID, address userAddress);
+error UserNotAuthenticated();
+error MetadataNotSupportedYet();
+
+abstract contract Verifier is IVerifier, ContextUpgradeable {
+    /// @dev Key to retrieve the linkID from the proof storage
+    string constant LINKED_PROOF_KEY = "linkID";
+
+    struct AuthTypeData {
+        IAuthValidator validator;
+        bytes params;
+        bool isActive;
+    }
+
+    struct UserAddressToIdInfo {
+        uint256 userID;
+        uint256 timestamp;
+    }
+
+    struct UserIdToAddressInfo {
+        address userAddress;
+        uint256 timestamp;
+    }
+
+    /// @custom:storage-location erc7201:iden3.storage.Verifier
+    struct VerifierStorage {
+        // Information about requests
+        // solhint-disable-next-line
+        mapping(uint256 requestId => mapping(address sender => VerifierLib.Proof[])) _proofs;
+        mapping(uint256 requestId => IVerifier.RequestData) _requests;
+        uint256[] _requestIds;
+        mapping(uint256 groupId => uint256[] requestIds) _groupedRequests;
+        uint256[] _groupIds;
+        IState _state;
+        // Information about multiRequests
+        mapping(uint256 multiRequestId => IVerifier.MultiRequest) _multiRequests;
+        uint256[] _multiRequestIds;
+        // Whitelisted validators
+        mapping(IRequestValidator => bool isApproved) _validatorWhitelist;
```

Comment:
The whitelisted validators mapping seem not to me used. The other mapping in the ValidatorWhitelist contract seem to be used.
If that's true than it can be removed

Reply:
Removed